### PR TITLE
Incorporated Conflict Modal from Gym, updates needed

### PR DESCRIPTION
**Pull request by YongbinWang**
Modal_newbookingpage

### CODE REVIEW

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (rldefa) [inline frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:8]: Remove todos if already done

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (rldefa) [inline frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:15]: Does this work? Pretty sure need to dispatch.

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (rldefa) [inline frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:35]: What is this meant to do?

**Review by rldefa — COMMENTED** [review]
Can there be a slight gap between the two buttons as in the design?

Commenting instead of approving because unsure if some css conform to our current style

**Comment by cjunxiang** [comment]
can ask designer to design the modal plz. HAHAH the spacings & textButton stylings v off

File: frontend/src/components/ConfirmationModal.tsx
Comment (hui444) [inline frontend/src/components/ConfirmationModal.tsx:6]: is this used?

File: frontend/src/components/ConfirmationModal.tsx
Comment (hui444) [inline frontend/src/components/ConfirmationModal.tsx:5]: can we order these better hahah
```suggestion
import React from 'react'
import { useHistory } from 'react-router-dom'

import styled from 'styled-components'
import Button from './Button'
import { PATHS } from '../routes/Routes'
```

File: frontend/src/components/ConfirmationModal.tsx
Comment (hui444) [inline frontend/src/components/ConfirmationModal.tsx:22]: might want to add a max width so modal does not get too big

File: frontend/src/components/ConfirmationModal.tsx
Comment (hui444) [inline frontend/src/components/ConfirmationModal.tsx:43]: not a valid font weight and size;
font weight by default is 400 so its not necessary. font-style is default normal too

File: frontend/src/components/ConfirmationModal.tsx
Comment (hui444) [inline frontend/src/components/ConfirmationModal.tsx:60]: is there a reason why this value is 16.8?

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (hui444) [inline frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:7]: use better name, like modalIsOpen, setModalIsOpen; include state! 
```suggestion
  const [modalIsOpen, setmodalIsOpen] = useState<boolean>(false)
```

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (hui444) [inline frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:9]: toggleStatus might be a better name for this, though not sure if theres any value in abstracting this into a function

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (hui444) [inline frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:13]: is update to make the additional call to book? kiv prince's pr..

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (hui444) [reply frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:15]: yup, need to update after prince finishes..

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (hui444) [inline frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:21]: button types should not be set here, confirmation modal should set them already since they are always going to be the same (right - primary, left - secondary)

File: frontend/src/components/ConfirmationModal.tsx
Comment (hui444) [inline frontend/src/components/ConfirmationModal.tsx:116]: does conflict link do anything onclick?

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (hui444) [inline frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:30]: requires text for link and actual link to call when it is clicked

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (hui444) [reply frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:35]: i think this was accidentally generated? not too sure.

**Review by hui444 — CHANGES_REQUESTED** [review]
pls fix changes! try and copy relevant stylings from figma instead of all of them !

File: frontend/src/components/ConfirmationModal.tsx
Comment (YongbinWang) [reply frontend/src/components/ConfirmationModal.tsx]: Resolved in commit 509ab25176f49a4a836ede036eb3559e1af54833.

File: frontend/src/components/ConfirmationModal.tsx
Comment (YongbinWang) [reply frontend/src/components/ConfirmationModal.tsx]: Resolved in commit 509ab25176f49a4a836ede036eb3559e1af54833.

File: frontend/src/components/ConfirmationModal.tsx
Comment (YongbinWang) [reply frontend/src/components/ConfirmationModal.tsx:21]: Resolved in commit 509ab25176f49a4a836ede036eb3559e1af54833.

File: frontend/src/components/ConfirmationModal.tsx
Comment (YongbinWang) [reply frontend/src/components/ConfirmationModal.tsx]: Resolved in commit 509ab25176f49a4a836ede036eb3559e1af54833.

File: frontend/src/components/ConfirmationModal.tsx
Comment (YongbinWang) [reply frontend/src/components/ConfirmationModal.tsx]: Following Figma styling.

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (YongbinWang) [reply frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx]: Resolved in commit 509ab25176f49a4a836ede036eb3559e1af54833.

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (YongbinWang) [reply frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx]: Resolved in commit 509ab25176f49a4a836ede036eb3559e1af54833.

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (YongbinWang) [reply frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:16]: Yes.

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (YongbinWang) [reply frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx]: Resolved in commit 509ab25176f49a4a836ede036eb3559e1af54833.

File: frontend/src/components/ConfirmationModal.tsx
Comment (YongbinWang) [reply frontend/src/components/ConfirmationModal.tsx]: Resolved in commit 509ab25176f49a4a836ede036eb3559e1af54833. Added conflict link to ViewConflict page.

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (YongbinWang) [reply frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:35]: Resolved in commit 509ab25176f49a4a836ede036eb3559e1af54833.

File: frontend/src/components/ConfirmationModal.tsx
Comment (hui444) [reply frontend/src/components/ConfirmationModal.tsx:21]: max width should be an absolute value, if not width is not restricted

File: frontend/src/components/ConfirmationModal.tsx
Comment (hui444) [inline frontend/src/components/ConfirmationModal.tsx:106]: might want to make the leftButtonType optional and default secondary

File: frontend/src/components/ModalLink.tsx
Comment (hui444) [inline frontend/src/components/ModalLink.tsx:2]: was this used?

File: frontend/src/components/ModalLink.tsx
Comment (hui444) [inline frontend/src/components/ModalLink.tsx:23]: this could be done in button; as a tertiary button - ie if state = 'tertiary', this is the button that renders

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (hui444) [inline frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:8]: might want to write this exactly? since there is only 1 prop
```suggestion
export default function ConflictBookingModal({booking}: { booking: Booking | undefined }) {
```

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (hui444) [reply frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:35]: use camel case! onLinkClick instead of onlinkClick

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (hui444) [reply frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx:22]: remove comment

File: frontend/src/store/facilityBooking/action.ts
Comment (hui444) [inline frontend/src/store/facilityBooking/action.ts:406]: remove comments!

File: frontend/src/store/facilityBooking/action.ts
Comment (hui444) [inline frontend/src/store/facilityBooking/action.ts:413]: remove these values after testing

File: frontend/src/store/facilityBooking/action.ts
Comment (hui444) [reply frontend/src/store/facilityBooking/action.ts:406]: this should be uncommented, remove the next const

File: frontend/src/store/facilityBooking/types.ts
Comment (hui444) [inline frontend/src/store/facilityBooking/types.ts:96]: are these necessary?

Comment (YongbinWang) [reply frontend/src/store/facilityBooking/types.ts]: Resolved in commit ca5730df3da1e495b5afa9c949a610730cb97605.

File: frontend/src/store/facilityBooking/action.ts
Comment (YongbinWang) [reply frontend/src/store/facilityBooking/action.ts]: Resolved in commit ca5730df3da1e495b5afa9c949a610730cb97605.

File: frontend/src/store/facilityBooking/action.ts
Comment (YongbinWang) [reply frontend/src/store/facilityBooking/action.ts]: Resolved in commit ca5730df3da1e495b5afa9c949a610730cb97605.

File: frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx
Comment (YongbinWang) [reply frontend/src/routes/FacilityBooking/ViewConflicts/ConflictBookingModal.tsx]: Resolved in commit f24c30e8b814ee8bc691b9803db23a6fc53234a6.

File: frontend/src/components/ModalLink.tsx
Comment (YongbinWang) [reply frontend/src/components/ModalLink.tsx:21]: Resolved in commit f24c30e8b814ee8bc691b9803db23a6fc53234a6.

File: frontend/src/components/ModalLink.tsx
Comment (YongbinWang) [reply frontend/src/components/ModalLink.tsx]: Resolved in commit f24c30e8b814ee8bc691b9803db23a6fc53234a6.

File: frontend/src/components/ConfirmationModal.tsx
Comment (YongbinWang) [reply frontend/src/components/ConfirmationModal.tsx:106]: Resolved in commit f24c30e8b814ee8bc691b9803db23a6fc53234a6.

File: frontend/src/components/ConfirmationModal.tsx
Comment (hui444) [inline frontend/src/components/ConfirmationModal.tsx:62]: not sure if this is necessary? if it can only be of 1 type, would it be better to not require leftButtonType at all?

File: frontend/src/routes/FacilityBooking/CreateBooking/index.tsx
Comment (hui444) [inline frontend/src/routes/FacilityBooking/CreateBooking/index.tsx:254]: can help change ValidForm() to something like formIsValid()? also there is no need to change between primary and secondary. primary would be sufficient

**Review by hui444 — APPROVED** [review]
pls fix all comments before merging

File: frontend/src/components/ConfirmationModal.tsx
Comment (YongbinWang) [reply frontend/src/components/ConfirmationModal.tsx]: Resolved in commit 8a39ccd6e0e18c9e6e1d9c931c1355c8e549ffdb.

File: frontend/src/routes/FacilityBooking/CreateBooking/index.tsx
Comment (YongbinWang) [reply frontend/src/routes/FacilityBooking/CreateBooking/index.tsx:254]: Resolved in commit 8a39ccd6e0e18c9e6e1d9c931c1355c8e549ffdb.

**Pull request merged**
